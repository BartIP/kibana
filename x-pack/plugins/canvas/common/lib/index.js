/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

export * from './datatable';
<<<<<<< HEAD
export * from './arg';
export * from './ast';
=======
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
export * from './autocomplete';
export * from './constants';
export * from './dataurl';
export * from './errors';
export * from './expression_form_handlers';
export * from './fetch';
export * from './find_in_object';
<<<<<<< HEAD
export * from './fn';
export * from './fonts';
export * from './functions_registry';
export * from './get_by_alias';
export * from './get_colors_from_palette';
export * from './get_field_type';
export * from './get_legend_config';
export * from './get_type';
export * from './grammar';
=======
export * from './fonts';
export * from './get_colors_from_palette';
export * from './get_field_type';
export * from './get_legend_config';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
export * from './handlebars';
export * from './hex_to_rgb';
export * from './httpurl';
export * from './latest_change';
export * from './missing_asset';
export * from './palettes';
export * from './pivot_object_array';
<<<<<<< HEAD
export * from './registry';
export * from './resolve_dataurl';
export * from './serialize';
export * from './type';
export * from './types_registry';
=======
export * from './resolve_dataurl';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
export * from './unquote_string';
export * from './url';
