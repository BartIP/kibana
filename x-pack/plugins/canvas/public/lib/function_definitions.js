/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import uniqBy from 'lodash.uniqby';
<<<<<<< HEAD
import { getServerFunctions } from '../state/selectors/app';
import { getBrowserRegistries } from './browser_registries';
=======
import { getBrowserRegistries } from '@kbn/interpreter/public/browser_registries';
import { getServerFunctions } from '../state/selectors/app';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

export async function getFunctionDefinitions(state) {
  const { browserFunctions } = await getBrowserRegistries();
  const serverFunctions = getServerFunctions(state);
  return uniqBy(serverFunctions.concat(browserFunctions.toArray()), 'name');
}
