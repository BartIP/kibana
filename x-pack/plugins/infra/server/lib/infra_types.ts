/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { InfraConfigurationAdapter } from './adapters/configuration';
import { InfraBackendFrameworkAdapter, InfraFrameworkRequest } from './adapters/framework';
<<<<<<< HEAD
import { InfraCapabilitiesDomain } from './domains/capabilities_domain';
import { InfraFieldsDomain } from './domains/fields_domain';
import { InfraLogEntriesDomain } from './domains/log_entries_domain';
=======
import { InfraFieldsDomain } from './domains/fields_domain';
import { InfraLogEntriesDomain } from './domains/log_entries_domain';
import { InfraMetadataDomain } from './domains/metadata_domain';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import { InfraMetricsDomain } from './domains/metrics_domain';
import { InfraNodesDomain } from './domains/nodes_domain';
import { InfraSourceStatus } from './source_status';
import { InfraSourceConfigurations, InfraSources } from './sources';

export interface InfraDomainLibs {
<<<<<<< HEAD
  capabilities: InfraCapabilitiesDomain;
=======
  metadata: InfraMetadataDomain;
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  fields: InfraFieldsDomain;
  logEntries: InfraLogEntriesDomain;
  nodes: InfraNodesDomain;
  metrics: InfraMetricsDomain;
}

export interface InfraBackendLibs extends InfraDomainLibs {
  configuration: InfraConfigurationAdapter<InfraConfiguration>;
  framework: InfraBackendFrameworkAdapter;
  sources: InfraSources;
  sourceStatus: InfraSourceStatus;
}

export interface InfraConfiguration {
  enabled: boolean;
  query: {
    partitionSize: number;
    partitionFactor: number;
  };
  sources: InfraSourceConfigurations;
}

export interface InfraContext {
  req: InfraFrameworkRequest;
}
