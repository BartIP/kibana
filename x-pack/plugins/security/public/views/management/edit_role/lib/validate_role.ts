/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

/*
 * Copyright Elasticsearch B.V. ../../../../../common/model/index_privileger one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

<<<<<<< HEAD
=======
import { i18n } from '@kbn/i18n';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import { IndexPrivilege } from '../../../../../common/model/index_privilege';
import { KibanaPrivilege } from '../../../../../common/model/kibana_privilege';
import { Role } from '../../../../../common/model/role';

/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

interface RoleValidatorOptions {
  shouldValidate?: boolean;
}

export interface RoleValidationResult {
  isInvalid: boolean;
  error?: string;
}

export class RoleValidator {
  private shouldValidate?: boolean;

  private inProgressSpacePrivileges: any[] = [];

  constructor(options: RoleValidatorOptions = {}) {
    this.shouldValidate = options.shouldValidate;
  }

  public enableValidation() {
    this.shouldValidate = true;
  }

  public disableValidation() {
    this.shouldValidate = false;
  }

  public validateRoleName(role: Role): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    if (!role.name) {
<<<<<<< HEAD
      return invalid(`Please provide a role name`);
    }
    if (role.name.length > 1024) {
      return invalid(`Name must not exceed 1024 characters`);
    }
    if (!role.name.match(/^[a-zA-Z_][a-zA-Z0-9_@\-\$\.]*$/)) {
      return invalid(
        `Name must begin with a letter or underscore and contain only letters, underscores, and numbers.`
=======
      return invalid(
        i18n.translate(
          'xpack.security.management.editRoles.validateRole.provideRoleNameWarningMessage',
          {
            defaultMessage: 'Please provide a role name',
          }
        )
      );
    }
    if (role.name.length > 1024) {
      return invalid(
        i18n.translate(
          'xpack.security.management.editRoles.validateRole.nameLengthWarningMessage',
          {
            defaultMessage: 'Name must not exceed 1024 characters',
          }
        )
      );
    }
    if (!role.name.match(/^[a-zA-Z_][a-zA-Z0-9_@\-\$\.]*$/)) {
      return invalid(
        i18n.translate(
          'xpack.security.management.editRoles.validateRole.nameAllowedCharactersWarningMessage',
          {
            defaultMessage:
              'Name must begin with a letter or underscore and contain only letters, underscores, and numbers.',
          }
        )
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      );
    }
    return valid();
  }

  public validateIndexPrivileges(role: Role): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    if (!Array.isArray(role.elasticsearch.indices)) {
<<<<<<< HEAD
      throw new TypeError(`Expected role.elasticsearch.indices to be an array`);
=======
      throw new TypeError(
        i18n.translate('xpack.security.management.editRoles.validateRole.indicesTypeErrorMessage', {
          defaultMessage: 'Expected {elasticIndices} to be an array',
          values: {
            elasticIndices: '"role.elasticsearch.indices"',
          },
        })
      );
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
    }

    const areIndicesValid =
      role.elasticsearch.indices
        .map(indexPriv => this.validateIndexPrivilege(indexPriv))
        .find((result: RoleValidationResult) => result.isInvalid) == null;

    if (areIndicesValid) {
      return valid();
    }
    return invalid();
  }

  public validateIndexPrivilege(indexPrivilege: IndexPrivilege): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    if (indexPrivilege.names.length && !indexPrivilege.privileges.length) {
<<<<<<< HEAD
      return invalid(`At least one privilege is required`);
=======
      return invalid(
        i18n.translate(
          'xpack.security.management.editRoles.validateRole.onePrivilegeRequiredWarningMessage',
          {
            defaultMessage: 'At least one privilege is required',
          }
        )
      );
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
    }
    return valid();
  }

  public validateSelectedSpaces(
    spaceIds: string[],
    privilege: KibanaPrivilege | null
  ): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    // If no assigned privilege, then no spaces are OK
    if (!privilege) {
      return valid();
    }

    if (Array.isArray(spaceIds) && spaceIds.length > 0) {
      return valid();
    }
<<<<<<< HEAD
    return invalid('At least one space is required');
=======
    return invalid(
      i18n.translate(
        'xpack.security.management.editRoles.validateRole.oneSpaceRequiredWarningMessage',
        {
          defaultMessage: 'At least one space is required',
        }
      )
    );
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  }

  public validateSelectedPrivilege(
    spaceIds: string[],
    privilege: KibanaPrivilege | null
  ): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    // If no assigned spaces, then a missing privilege is OK
    if (!spaceIds || spaceIds.length === 0) {
      return valid();
    }

    if (privilege) {
      return valid();
    }
<<<<<<< HEAD
    return invalid('Privilege is required');
=======
    return invalid(
      i18n.translate(
        'xpack.security.management.editRoles.validateRole.privilegeRequiredWarningMessage',
        {
          defaultMessage: 'Privilege is required',
        }
      )
    );
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  }

  public setInProgressSpacePrivileges(inProgressSpacePrivileges: any[]) {
    this.inProgressSpacePrivileges = [...inProgressSpacePrivileges];
  }

  public validateInProgressSpacePrivileges(role: Role): RoleValidationResult {
    const { global } = role.kibana;

    // A Global privilege of "all" will ignore all in progress privileges,
    // so the form should not block saving in this scenario.
    const shouldValidate = this.shouldValidate && !global.includes('all');

    if (!shouldValidate) {
      return valid();
    }

    const allInProgressValid = this.inProgressSpacePrivileges.every(({ spaces, privilege }) => {
      return (
        !this.validateSelectedSpaces(spaces, privilege).isInvalid &&
        !this.validateSelectedPrivilege(spaces, privilege).isInvalid
      );
    });

    if (allInProgressValid) {
      return valid();
    }
    return invalid();
  }

  public validateSpacePrivileges(role: Role): RoleValidationResult {
    if (!this.shouldValidate) {
      return valid();
    }

    const privileges = Object.values(role.kibana.space || {});

    const arePrivilegesValid = privileges.every(assignedPrivilege => !!assignedPrivilege);
    const areInProgressPrivilegesValid = !this.validateInProgressSpacePrivileges(role).isInvalid;

    if (arePrivilegesValid && areInProgressPrivilegesValid) {
      return valid();
    }
    return invalid();
  }

  public validateForSave(role: Role): RoleValidationResult {
    const { isInvalid: isNameInvalid } = this.validateRoleName(role);
    const { isInvalid: areIndicesInvalid } = this.validateIndexPrivileges(role);
    const { isInvalid: areSpacePrivilegesInvalid } = this.validateSpacePrivileges(role);

    if (isNameInvalid || areIndicesInvalid || areSpacePrivilegesInvalid) {
      return invalid();
    }

    return valid();
  }
}

function invalid(error?: string): RoleValidationResult {
  return {
    isInvalid: true,
    error,
  };
}

function valid(): RoleValidationResult {
  return {
    isInvalid: false,
  };
}
