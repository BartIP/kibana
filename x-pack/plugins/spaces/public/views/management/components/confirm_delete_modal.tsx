/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import {
  EuiButton,
  EuiButtonEmpty,
  EuiCallOut,
  // @ts-ignore
  EuiConfirmModal,
  EuiFieldText,
  EuiFormRow,
  EuiModal,
  EuiModalBody,
  EuiModalFooter,
  EuiModalHeader,
  EuiModalHeaderTitle,
  EuiOverlayMask,
  EuiText,
} from '@elastic/eui';
<<<<<<< HEAD
=======
import { FormattedMessage, InjectedIntl, injectI18n } from '@kbn/i18n/react';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import { SpacesNavState } from 'plugins/spaces/views/nav_control';
import React, { ChangeEvent, Component } from 'react';
import { Space } from '../../../../common/model/space';
import { SpacesManager } from '../../../lib';

interface Props {
  space: Space;
  spacesManager: SpacesManager;
  spacesNavState: SpacesNavState;
  onCancel: () => void;
  onConfirm: () => void;
<<<<<<< HEAD
=======
  intl: InjectedIntl;
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
}

interface State {
  confirmSpaceName: string;
  error: boolean | null;
  deleteInProgress: boolean;
}

<<<<<<< HEAD
export class ConfirmDeleteModal extends Component<Props, State> {
=======
class ConfirmDeleteModalUI extends Component<Props, State> {
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
  public state = {
    confirmSpaceName: '',
    error: null,
    deleteInProgress: false,
  };

  public render() {
<<<<<<< HEAD
    const { space, spacesNavState, onCancel } = this.props;
=======
    const { space, spacesNavState, onCancel, intl } = this.props;
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1

    let warning = null;
    if (isDeletingCurrentSpace(space, spacesNavState)) {
      const name = (
        <span>
          (<strong>{space.name}</strong>)
        </span>
      );
      warning = (
        <EuiCallOut color="warning">
          <EuiText>
<<<<<<< HEAD
            You are about to delete your current space {name}. You will be redirected to choose a
            different space if you continue.
=======
            <FormattedMessage
              id="xpack.spaces.management.confirmDeleteModal.redirectAfterDeletingCurrentSpaceWarningMessage"
              defaultMessage="You are about to delete your current space {name}. You will be redirected to choose a different space if you continue."
              values={{ name }}
            />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
          </EuiText>
        </EuiCallOut>
      );
    }

    // This is largely the same as the built-in EuiConfirmModal component, but we needed the ability
    // to disable the buttons since this could be a long-running operation

    return (
      <EuiOverlayMask>
        <EuiModal onClose={onCancel} className={'spcConfirmDeleteModal'}>
          <EuiModalHeader>
            <EuiModalHeaderTitle data-test-subj="confirmModalTitleText">
<<<<<<< HEAD
              Delete space {`'${space.name}'`}
=======
              <FormattedMessage
                id="xpack.spaces.management.confirmDeleteModal.confirmDeleteSpaceButtonLabel"
                defaultMessage="Delete space {spaceName}"
                values={{
                  spaceName: "'" + space.name + "'",
                }}
              />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
            </EuiModalHeaderTitle>
          </EuiModalHeader>
          <EuiModalBody>
            <EuiText data-test-subj="confirmModalBodyText">
              <p>
<<<<<<< HEAD
                Deleting a space permanently removes the space and{' '}
                <strong>all of its contents</strong>. You can't undo this action.
              </p>

              <EuiFormRow
                label={'Confirm space name'}
                isInvalid={!!this.state.error}
                error={'Space names do not match.'}
=======
                <FormattedMessage
                  id="xpack.spaces.management.confirmDeleteModal.deletingSpaceWarningMessage"
                  defaultMessage="Deleting a space permanently removes the space and {allContents}. You can't undo this action."
                  values={{
                    allContents: (
                      <strong>
                        <FormattedMessage
                          id="xpack.spaces.management.confirmDeleteModal.allContentsText"
                          defaultMessage="all of its contents"
                        />
                      </strong>
                    ),
                  }}
                />
              </p>

              <EuiFormRow
                label={intl.formatMessage({
                  id: 'xpack.spaces.management.confirmDeleteModal.confirmSpaceNameFormRowLabel',
                  defaultMessage: 'Confirm space name',
                })}
                isInvalid={!!this.state.error}
                error={intl.formatMessage({
                  id: 'xpack.spaces.management.confirmDeleteModal.spaceNamesDoNoMatchErrorMessage',
                  defaultMessage: 'Space names do not match.',
                })}
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
              >
                <EuiFieldText
                  value={this.state.confirmSpaceName}
                  onChange={this.onSpaceNameChange}
                  disabled={this.state.deleteInProgress}
                />
              </EuiFormRow>

              {warning}
            </EuiText>
          </EuiModalBody>
          <EuiModalFooter>
            <EuiButtonEmpty
              data-test-subj="confirmModalCancelButton"
              onClick={onCancel}
              isDisabled={this.state.deleteInProgress}
            >
<<<<<<< HEAD
              Cancel
=======
              <FormattedMessage
                id="xpack.spaces.management.confirmDeleteModal.cancelButtonLabel"
                defaultMessage="Cancel"
              />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
            </EuiButtonEmpty>

            <EuiButton
              data-test-subj="confirmModalConfirmButton"
              onClick={this.onConfirm}
              fill
              color={'danger'}
              isLoading={this.state.deleteInProgress}
            >
<<<<<<< HEAD
              Delete space and all contents
=======
              <FormattedMessage
                id="xpack.spaces.management.confirmDeleteModal.deleteSpaceAndAllContentsButtonLabel"
                defaultMessage=" Delete space and all contents"
              />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
            </EuiButton>
          </EuiModalFooter>
        </EuiModal>
      </EuiOverlayMask>
    );
  }

  private onSpaceNameChange = (e: ChangeEvent<HTMLInputElement>) => {
    if (typeof this.state.error === 'boolean') {
      this.setState({
        confirmSpaceName: e.target.value,
        error: e.target.value !== this.props.space.name,
      });
    } else {
      this.setState({
        confirmSpaceName: e.target.value,
      });
    }
  };

  private onConfirm = async () => {
    if (this.state.confirmSpaceName === this.props.space.name) {
      const needsRedirect = isDeletingCurrentSpace(this.props.space, this.props.spacesNavState);
      const spacesManager = this.props.spacesManager;

      this.setState({
        deleteInProgress: true,
      });

      await this.props.onConfirm();

      this.setState({
        deleteInProgress: false,
      });

      if (needsRedirect) {
        spacesManager.redirectToSpaceSelector();
      }
    } else {
      this.setState({
        error: true,
      });
    }
  };
}

function isDeletingCurrentSpace(space: Space, spacesNavState: SpacesNavState) {
  return space.id === spacesNavState.getActiveSpace().id;
}
<<<<<<< HEAD
=======

export const ConfirmDeleteModal = injectI18n(ConfirmDeleteModalUI);
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
