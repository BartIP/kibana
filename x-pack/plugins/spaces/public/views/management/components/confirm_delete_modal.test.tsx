/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */
<<<<<<< HEAD
import { mount, shallow } from 'enzyme';
import React from 'react';
=======
import React from 'react';
import { mountWithIntl, shallowWithIntl } from 'test_utils/enzyme_helpers';
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
import { SpacesManager } from '../../../lib';
import { SpacesNavState } from '../../nav_control';
import { ConfirmDeleteModal } from './confirm_delete_modal';

const buildMockChrome = () => {
  return {
    addBasePath: (path: string) => path,
  };
};

describe('ConfirmDeleteModal', () => {
  it('renders as expected', () => {
    const space = {
      id: 'my-space',
      name: 'My Space',
    };

    const mockHttp = {
      delete: jest.fn(() => Promise.resolve()),
    };
    const mockChrome = buildMockChrome();

    const spacesManager = new SpacesManager(mockHttp, mockChrome, '/');

    const spacesNavState: SpacesNavState = {
      getActiveSpace: () => space,
      refreshSpacesList: jest.fn(),
    };

    const onCancel = jest.fn();
    const onConfirm = jest.fn();

    expect(
<<<<<<< HEAD
      shallow(
        <ConfirmDeleteModal
=======
      shallowWithIntl(
        <ConfirmDeleteModal.WrappedComponent
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
          space={space}
          spacesManager={spacesManager}
          spacesNavState={spacesNavState}
          onCancel={onCancel}
          onConfirm={onConfirm}
<<<<<<< HEAD
=======
          intl={null as any}
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
        />
      )
    ).toMatchSnapshot();
  });

  it(`requires the space name to be typed before confirming`, () => {
    const space = {
      id: 'my-space',
      name: 'My Space',
    };

    const mockHttp = {
      delete: jest.fn(() => Promise.resolve()),
    };
    const mockChrome = buildMockChrome();

    const spacesManager = new SpacesManager(mockHttp, mockChrome, '/');

    const spacesNavState: SpacesNavState = {
      getActiveSpace: () => space,
      refreshSpacesList: jest.fn(),
    };

    const onCancel = jest.fn();
    const onConfirm = jest.fn();

<<<<<<< HEAD
    const wrapper = mount(
      <ConfirmDeleteModal
=======
    const wrapper = mountWithIntl(
      <ConfirmDeleteModal.WrappedComponent
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
        space={space}
        spacesManager={spacesManager}
        spacesNavState={spacesNavState}
        onCancel={onCancel}
        onConfirm={onConfirm}
<<<<<<< HEAD
=======
        intl={null as any}
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      />
    );

    const input = wrapper.find('input');
    expect(input).toHaveLength(1);

    input.simulate('change', { target: { value: 'My Invalid Space Name ' } });

    const confirmButton = wrapper.find('button[data-test-subj="confirmModalConfirmButton"]');
    confirmButton.simulate('click');

    expect(onConfirm).not.toHaveBeenCalled();

    input.simulate('change', { target: { value: 'My Space' } });
    confirmButton.simulate('click');

    expect(onConfirm).toHaveBeenCalledTimes(1);
  });
});
