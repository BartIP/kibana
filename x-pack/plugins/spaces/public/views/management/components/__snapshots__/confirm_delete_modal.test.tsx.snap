// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDeleteModal renders as expected 1`] = `
<EuiOverlayMask>
  <EuiModal
    className="spcConfirmDeleteModal"
    maxWidth={true}
    onClose={[MockFunction]}
  >
    <EuiModalHeader>
      <EuiModalHeaderTitle
        data-test-subj="confirmModalTitleText"
      >
<<<<<<< HEAD
        Delete space 
        'My Space'
=======
        <FormattedMessage
          defaultMessage="Delete space {spaceName}"
          id="xpack.spaces.management.confirmDeleteModal.confirmDeleteSpaceButtonLabel"
          values={
            Object {
              "spaceName": "'My Space'",
            }
          }
        />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      </EuiModalHeaderTitle>
    </EuiModalHeader>
    <EuiModalBody>
      <EuiText
        data-test-subj="confirmModalBodyText"
        grow={true}
        size="m"
      >
        <p>
<<<<<<< HEAD
          Deleting a space permanently removes the space and
           
          <strong>
            all of its contents
          </strong>
          . You can't undo this action.
=======
          <FormattedMessage
            defaultMessage="Deleting a space permanently removes the space and {allContents}. You can't undo this action."
            id="xpack.spaces.management.confirmDeleteModal.deletingSpaceWarningMessage"
            values={
              Object {
                "allContents": <strong>
                  <FormattedMessage
                    defaultMessage="all of its contents"
                    id="xpack.spaces.management.confirmDeleteModal.allContentsText"
                    values={Object {}}
                  />
                </strong>,
              }
            }
          />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
        </p>
        <EuiFormRow
          describedByIds={Array []}
          error="Space names do not match."
          fullWidth={false}
          hasEmptyLabelSpace={false}
          isInvalid={false}
          label="Confirm space name"
        >
          <EuiFieldText
            compressed={false}
            disabled={false}
            fullWidth={false}
            isLoading={false}
            onChange={[Function]}
            value=""
          />
        </EuiFormRow>
        <EuiCallOut
          color="warning"
          size="m"
        >
          <EuiText
            grow={true}
            size="m"
          >
<<<<<<< HEAD
            You are about to delete your current space 
            <span>
              (
              <strong>
                My Space
              </strong>
              )
            </span>
            . You will be redirected to choose a different space if you continue.
=======
            <FormattedMessage
              defaultMessage="You are about to delete your current space {name}. You will be redirected to choose a different space if you continue."
              id="xpack.spaces.management.confirmDeleteModal.redirectAfterDeletingCurrentSpaceWarningMessage"
              values={
                Object {
                  "name": <span>
                    (
                    <strong>
                      My Space
                    </strong>
                    )
                  </span>,
                }
              }
            />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
          </EuiText>
        </EuiCallOut>
      </EuiText>
    </EuiModalBody>
    <EuiModalFooter>
      <EuiButtonEmpty
        color="primary"
        data-test-subj="confirmModalCancelButton"
        iconSide="left"
        isDisabled={false}
        onClick={[MockFunction]}
        type="button"
      >
<<<<<<< HEAD
        Cancel
=======
        <FormattedMessage
          defaultMessage="Cancel"
          id="xpack.spaces.management.confirmDeleteModal.cancelButtonLabel"
          values={Object {}}
        />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      </EuiButtonEmpty>
      <EuiButton
        color="danger"
        data-test-subj="confirmModalConfirmButton"
        fill={true}
        iconSide="left"
        isLoading={false}
        onClick={[Function]}
        type="button"
      >
<<<<<<< HEAD
        Delete space and all contents
=======
        <FormattedMessage
          defaultMessage=" Delete space and all contents"
          id="xpack.spaces.management.confirmDeleteModal.deleteSpaceAndAllContentsButtonLabel"
          values={Object {}}
        />
>>>>>>> ff49a1c6742d67fa5daed569ff3bb269783f6bd1
      </EuiButton>
    </EuiModalFooter>
  </EuiModal>
</EuiOverlayMask>
`;
